packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -463,16 +462,15 @@ export class DotNetGenerator extends Generator {
   }
 
   private namespaceFor(assm: spec.Assembly, type: spec.Type): string {
-    const parts = [assm.targets!.dotnet!.namespace];
     let ns = type.namespace;
     while (ns != null && assm.types?.[`${assm.name}.${ns}`] != null) {
       const nesting = assm.types[`${assm.name}.${ns}`];
       ns = nesting.namespace;
     }
     if (ns != null) {
-      parts.push(...ns.split('.').map((n) => toPascalCase(n)));
+      return this.typeresolver.resolveNamespace(assm, assm.name, ns);
     }
-    return parts.join('.');
+    return assm.targets!.dotnet!.namespace;
```

Comment:
Can the 'non-nullability' of these fields not be represented in the assembly spec because of the presence of the dotnet target isn't guaranteed?

Reply:
Problem is the spec doesn't actually represent per-language configurations... so they're opaque stuff entirely.